api/views.py

```diff
@@ -0,0 +1,148 @@
+        return Response("Summary cost of all salaries: " + str(sum_of_salaries['salary__sum']))
+
+
+class IncomeViewSet(viewsets.ModelViewSet):
+    queryset = Income.objects.all()
+    serializer_class = IncomeSerializer
+    filter_fields = ('name', 'date')
+
+    @action(detail=False, methods=['get'])
+    def last_incomes(self, request, *args, **kwargs):
+        number_of_incomes_to_show = request.GET.get('number_of_incomes', None)
+        if number_of_incomes_to_show is not None:
+            incomes = Income.objects.order_by('date')[:int(number_of_incomes_to_show)]
+        else:
+            incomes = Income.objects.order_by('date')
+        serializer = EventSerializer(incomes, many=True)
+        return Response(serializer.data)
+
+    @action(detail=False, methods=['get'])
+    def sum_of_cost(self, request, *args, **kwargs):
+        days_back = request.GET.get('days', 7)
+        current_date = datetime.today().date()
+        date_for_filter = current_date - timedelta(days=int(days_back))
+        sum_of_incomes = Income.objects.filter(date__gte=date_for_filter).aggregate(Sum('sum'))
+        return Response("Total income from the last " + str(days_back) + " days is: " + str(sum_of_incomes['sum__sum']))
+
+
+class OutcomeViewSet(viewsets.ModelViewSet):
+    queryset = Outcome.objects.all()
+    serializer_class = OutcomeSerializer
+    filter_fields = ('name', 'date')
+
+    @action(detail=False, methods=['get'])
+    def last_outcomes(self, request, *args, **kwargs):
+        number_of_outcomes_to_show = request.GET.get('number_of_outcomes', None)
+        if number_of_outcomes_to_show is not None:
+            outcomes = Outcome.objects.order_by('date')[:int(number_of_outcomes_to_show)]
+        else:
+            outcomes = Outcome.objects.order_by('date')
+        serializer = EventSerializer(outcomes, many=True)
+        return Response(serializer.data)
+
+    @action(detail=False, methods=['get'])
+    def sum_of_cost(self, request, *args, **kwargs):
+        days_back = request.GET.get('days', 7)
+        current_date = datetime.today().date()
+        date_for_filter = current_date - timedelta(days=int(days_back))
+        sum_of_outcomes = Outcome.objects.filter(date__gte=date_for_filter).aggregate(Sum('sum'))
+        return Response(
+            "Total outcome from the last " + str(days_back) + " days is: " + str(sum_of_outcomes['sum__sum']))
+
+
+class EventViewSet(viewsets.ModelViewSet):
+    queryset = Event.objects.all()
+    serializer_class = EventMiniSerializer
+
+    @action(detail=False, methods=['get'])
+    def future_events(self, request, *args, **kwargs):
+        events = Event.objects.future_events()
+
+        serializer = EventSerializer(events, many=True)
+        return Response(serializer.data)
+
+    @action(detail=False, methods=['get'])
+    def next_event(self, request, *args, **kwargs):
+        event = Event.objects.next_event()
+
+        serializer = EventSerializer(event, many=False)
```

Comment:
`many=False` is the default value I think, but that's fine

Reply:
ok